src/Sentry/Internal/CachingBackgroundWorker.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Sentry.Extensibility;
+using Sentry.Protocol.Envelopes;
+
+namespace Sentry.Internal
+{
+    internal class CachingBackgroundWorker : BackgroundWorkerBase
```

Comment:
We're not going to be able to use this class later outside of the background thread..
What if we used composition here somehow?

The reason I ask is the Xamarin crash handler use case, where we need to write to disk but not sure we can do on a threadpool thread.

We can try that first before changing anything, and we can change it later too, just thinking ahead a bit

Reply:
We may want to consolidate into one approach somehow, because otherwise we'll have to have two caching background workers, one for Xamarin and one for everything else.

Reply:
I'll try your code now on Xamarin crash handler